packages/maskbook/src/plugins/NFT/UI/NFTInPost.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from 'react'
+import MaskbookPluginWrapper from '../../MaskbookPluginWrapper'
+import type { NFTInPostProps, TokenDetails } from '../types'
+import { parseNftUrl } from '../utils'
+import NFTCardUI from './NFTCardUI'
+
+export default function NFTInPost(props: NFTInPostProps) {
+    const [tokenDetails, setTokenDetails] = useState<TokenDetails>({})
+
+    useEffect(function () {
+        const [address, tokenId] = parseNftUrl(props.nftUrl)
+
+        if (address && tokenId) {
+            const link = new URL('https://api.opensea.io/api/v1/assets')
+            link.searchParams.set('asset_contract_address', address)
+            link.searchParams.set('token_ids', tokenId)
+
+            fetch(link.toString())
```

Comment:
Hi, I suggest you to use `Services.Helper.fetch`. In this environment, your request might be blocked by CSP. But you still need to care about CORS, `Services.Helper.fetch` is a "proxy" that forwards your request to the background page.